### PR TITLE
Update MailTemplate.php

### PR DESCRIPTION
**Pull request by rigin**
The method does not bind the template to the extension. And I propose to correct this omission.

Pull Request for Issue # .

### Summary of Changes
When using this method, the 'extension' field remains empty and, accordingly, the template is not visible in the list of templates.
I added a parameter to insert it.If it is not used, then the 'extension' field is filled with the contents of the 'name' field before the '.' separator.

### CODE REVIEW

File: libraries/src/Mail/MailTemplate.php [hunk 1]
```diff
@@ -411,17 +411,19 @@ public static function getTemplate($key, $language)
...
 	 * @since   4.0.0
 	 */
-	public static function createTemplate($key, $subject, $body, $tags, $htmlbody = '')
+	public static function createTemplate($key, $subject, $body, $tags, $htmlbody = '',$extension = null)
 	{
 		$db = Factory::getDbo();
 
 		$template = new \stdClass;
 		$template->template_id = $key;
+		$template->extension = $extension??(explode('.', $key)[0]);
```
Comment (richard67) [inline libraries/src/Mail/MailTemplate.php]:
```suggestion
		$template->extension = $extension ?? (explode('.', $key)[0]);
```
Fix failing PHP code style checks in drone ci.

File: libraries/src/Mail/MailTemplate.php [hunk 2]
```diff
@@ -411,17 +411,19 @@ public static function getTemplate($key, $language)
...
 	 * @param   array   $tags      Associative array of tags to replace
 	 * @param   string  $htmlbody  A default htmlbody (normally a translatable string)
+	 * @param   string  $extension The extension to which the template belongs
 	 *
 	 * @return  boolean  True on success, false on failure
 	 *
 	 * @since   4.0.0
 	 */
-	public static function createTemplate($key, $subject, $body, $tags, $htmlbody = '')
+	public static function createTemplate($key, $subject, $body, $tags, $htmlbody = '',$extension = null)
```
Comment (Quy) [inline libraries/src/Mail/MailTemplate.php]:
```suggestion
	public static function createTemplate($key, $subject, $body, $tags, $htmlbody = '', $extension = null)
```

**Comment by HLeithner** [comment]
Adding a parameter is a b/c break and can't go into a minor version.

**Comment by brianteeman** [comment]
> Adding a parameter is a b/c break and can't go into a minor version.

even when added at the end and with a default value?

**Comment by HLeithner** [comment]
This pull requests has automatically rebased to 4.2-dev.

**Review by Hackwar — APPROVED** [review]

**Comment by Fedik** [comment]
To keep backward compatibility we should extract extension from the `$key` for now.
Adding extra parameter can be done in 5.x version, or we should review how all this thing works.

**Review by Kaushik1216 — APPROVED** [review]

**Comment by ReLater** [comment]
@rigin
Because we have another pr (https://github.com/joomla/joomla-cms/pull/39540) of @Kaushik1216 just the question (no pushing!) if you will continue your pr here; whenever you'll have the time. Thank you!

**Comment by Quy** [comment]
Closing in favor of #39540. Thank you.

**Pull request closed without merging**